python_modules/dagster/dagster_tests/core_tests/test_asset_events.py

```diff
@@ -166,3 +171,103 @@ def my_other_asset(my_asset):
             )
         )
         assert record.event_log_entry.dagster_event.event_specific_data.partition is None
+
+
+def test_subset_on_asset_materialization_planned_event_for_single_run_backfill_allowed():
```

Comment:
Can you add tests flexing how some of the asset partition reads are affected?

`get_materialized_partitions`, `get_latest_storage_id_by_partition`, `get_latest_tags_by_partition`, `get_latest_asset_partition_materialization_attempts_without_materializations`

Reply:
I've added tests for these methods

Except `get_latest_tags_by_partition`, that method filters by event type and materialization planned events can't contain tags right now